src/ripple/ledger/impl/View.cpp

```diff
@@ -810,19 +805,28 @@ describeOwnerDir(AccountID const& account)
     };
 }
 
-std::pair<TER, bool>
+boost::optional<std::uint64_t>
 dirAdd (ApplyView& view,
-    std::uint64_t&                          uNodeDir,
     Keylet const&                           dir,
     uint256 const&                          uLedgerIndex,
+    bool                                    strictOrder,
     std::function<void (SLE::ref)>          fDescriber,
     beast::Journal j)
```

Comment:
We should consider splitting `dirAdd` into `dirAppend` and `dirInsert` as well. I don't like that the high-level interface still has the bool parameter, while the lower level one is actually cleaner. I'd have less objection if it were the other way around (i.e. a high level function then sets the correct flags to a lower level function).